uphold-integration/src/main/java/org/dash/wallet/integration/uphold/data/UpholdClient.java

```diff
@@ -236,7 +280,16 @@ public void onSuccess(String data) {
 
             @Override
             public void onError(Exception e, boolean otpRequired) {
-
+                log.error("Error loading Dash balance: " + e.getMessage());
+                if(e instanceof UpholdException) {
+                    UpholdException ue = (UpholdException)e;
+                    if(ue.getCode() >= 400) {
+                        //we don't have the correct access token, let's logout
+                        accessToken = null;
+                        storeAccessToken();
+                    }
```

Comment:
Error 401 is Unauthorized which is the result when the user changes the password in uphold and the app cannot link to the uphold account with its access token.

Here we are allowing any error greater than 400 to be used as a reason to reset the access token, which will require the user to relink their account.  We may want to limit this to 401 only.

Reply:
I also found that the getDashBalance is the first method called when the user accesses the Uphold screen after linking to their account.  If this fails with 401, then we can unlink it in the app, requiring the user to relink in the case of a changed password.